#!/usr/bin/env python
PACKAGE='btr_point_tracker'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                       Type       Reconfiguration levexl            Description                       Default   Min   Max
gen.add("verbose_mode", int_t,  -1 , "verbose_mode",  0, 0, 1)
gen.add("min_hessian",int_t , -1, "Minimum Hessian (SURF)", 250, 100, 1000)
gen.add("min_features",int_t , -1, "Minimum num features (ORB)", 30, 10, 100)
gen.add("max_correspondence_dist",int_t , -1, "Maximum distance between consecutive correspondences [pixels]", 20, 5, 100)
gen.add("feature_type",int_t , -1, "Type of feature. (1:SURF, 2:ORB, 3:BRISK", 2, 1, 3)

exit(gen.generate(PACKAGE, "btr_point_tracker_node", "tracker_params"))